annotated.go

```diff
@@ -89,3 +90,268 @@ func (a Annotated) String() string {
 	}
 	return fmt.Sprintf("fx.Annotated{%v}", strings.Join(fields, ", "))
 }
+
+type annotations struct {
+	ParamTags  []string
+	ResultTags []string
+}
+
+// Annotation can be passed to Annotate(f interface{}, anns ...Annotation)
+// for annotating the parameter and result types of a function.
+type Annotation interface {
+	apply(*annotations) error
+	getAnnotatedType(reflect.Type) []reflect.Type
+}
+
+type annotationError struct {
```

Comment:
Since this is used in a special way, consider adding some comments here to explain it.